### PR TITLE
Bring spanner home

### PR DESCRIPTION
**Pull request by christophermaier**
A big chunk of Spanner is now only used here in Cog; it should live in Cog.

### CODE REVIEW

File: lib/cog/gen_command.ex [hunk 1]
```diff
@@ -0,0 +1,251 @@
+defmodule Cog.GenCommand do
```
Comment (kevsmith) [inline lib/cog/gen_command.ex]: Should be nested under `Cog.Command`?

File: lib/cog/bundle/config.ex [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    invocation name, the Elixir module that implements it, and the
+    various options the command may take
+  - A list of permissions the bundle will create
+  - A list of initial rules for the commands in the bundle, using the
+    bundle permissions.
+
+  ## Example
+
+      %{bundle: %{name: "foo"},
+        commands: [%{module: "Spanner.Commands.AddRule",
```
Comment (vanstee) [inline lib/cog/bundle/config.ex]: Do we want to update these example modules? Might be confusing.

File: lib/cog/command/request.ex [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        {:ok, config}
+      {:error, error} ->
+        err = "Unable to read the command config file for the command '#{request.command}'. #{inspect error}"
+        Logger.error(err)
+        {:error, err}
+    end
+  end
+
+  defp open_config(request) do
+    case Application.get_env(:spanner, :command_config_root) do
```
Comment (vanstee) [inline lib/cog/command/request.ex]: Are we going to leave `:spanner` as the config for now?

**Comment by vanstee** [comment]
LGTM

File: lib/cog/bundle/config.ex (hunk 2 above)
Comment (christophermaier) [inline lib/cog/bundle/config.ex]: D'oh

**Comment by kevsmith** [comment]
👍

**Pull request merged**